conda/fetch.py

```diff
@@ -91,31 +82,48 @@ def fetch_repodata(url, cache_dir=None, use_cache=False, session=None):
     if use_cache:
         return cache
 
+    if not ssl_verify:
+        try:
+            from requests.packages.urllib3.connectionpool import InsecureRequestWarning
+        except ImportError:
+            pass
+        else:
+            warnings.simplefilter('ignore', InsecureRequestWarning)
+
+    session = session or CondaSession()
+
```

Comment:
^^ Just switched the ordering here.  Don't need this logic in front of `if use_cache: return cache`.